ui/src/app/pages/login/account-creation/account-creation-institution.spec.tsx

```diff
@@ -1,417 +1,332 @@
-}
-
-function getRoleDropdown(wrapper: AnyWrapper) {
-  const matchedElements = wrapper
-    .find('Dropdown[data-test-id="role-dropdown"]')
-    ?.getElements();
-
-  return matchedElements ? matchedElements[0] : undefined;
-}
-
-function getSubmitButton(wrapper: AnyWrapper): AnyWrapper {
-  return wrapper.find('[data-test-id="submit-button"]');
-}
-
-const academicSpecificRoleOption =
-  AccountCreationOptions.institutionalRoleOptions.find(
-    (x) => x.value === InstitutionalRole.UNDERGRADUATE
-  );
-
-const industrySpecificRoleOption =
-  AccountCreationOptions.institutionalRoleOptions.find(
-    (x) => x.value === InstitutionalRole.SENIOR_RESEARCHER
-  );
-
-const eventDefaults = {
-  stopPropagation: () => undefined,
-  preventDefault: () => undefined,
+const profile: Profile = {
+  generalDiscoverySources: undefined,
+  partnerDiscoverySources: undefined,
+  username: '',
+  contactEmail: 'contactEmail@broadinstitute.org',
+  verifiedInstitutionalAffiliation: {
+    institutionShortName: BROAD.shortName,
+    institutionDisplayName: BROAD.displayName,
+    institutionalRoleEnum: InstitutionalRole.FELLOW,
+  },
 };
-
-beforeEach(() => {
-  serverConfigStore.set({ config: defaultServerConfig });
-  registerApiClient(ConfigApi, new ConfigApiStub());
-  registerApiClient(InstitutionApi, new InstitutionApiStub());
-
-  props = {
-    profile: createEmptyProfile(),
-    onComplete: () => {},
-    onPreviousClick: () => {},
+const setup = (profileArg = profile) => {
+  return {
+    container: render(
+      <AccountCreationInstitution
+        profile={profileArg}
+        onComplete={() => {}}
+        onPreviousClick={() => {}}
+      />
+    ).container,
+    user: userEvent.setup(),
   };
+};
 
-  mockGetPublicInstitutionDetails = jest.spyOn(
-    institutionApi(),
-    'getPublicInstitutionDetails'
-  );
-});
-
-it('should render', async () => {
-  const wrapper = component();
-  expect(wrapper.exists()).toBeTruthy();
-});
-
-it('should load institutions list', async () => {
-  const wrapper = component();
-  await waitOneTickAndUpdate(wrapper);
-
-  expect(mockGetPublicInstitutionDetails).toHaveBeenCalled();
-
-  const options = getInstitutionDropdown(wrapper).props.options as Array<{
-    value;
-    label;
-  }>;
-  expect(options.length).toEqual(defaultInstitutions.length);
-  // Drop down list should be sorted in ASC order
-  expect(options[0].label).not.toEqual(defaultInstitutions[0].displayName);
-  expect(options[0].label).toEqual('Broad Institute');
-  console.log(options[0]);
-  console.log(defaultInstitutions[0]);
-});
-
-it('should show user-facing error message on data load error', async () => {
-  mockGetPublicInstitutionDetails.mockRejectedValueOnce(
-    new Response(null, { status: 500 })
+const waitForSpinnerToGoAway = async (container) => {
+  await waitFor(() =>
+    container.querySelector(
+      '#account-creation-institution > div:nth-child(2) > div:nth-child(1) > div:nth-child(4) > div > svg'
```

Comment:
Is there a more direct way of finding the spinner? I feel like this might end up changing without us expecting it to.